scenario/objectsearch.go

```diff
@@ -0,0 +1,369 @@
+		return nil, errors.Errorf("Unknown ObjectSearchSource<%d>", value)
+	}
+	return []byte(fmt.Sprintf(`"%s"`, str)), nil
+}
+
+// String representation of ObjectSearchType
+func (value ObjectSearchType) String() string {
+	sType, err := objectSearchTypeEnumMap.String(int(value))
+	if err != nil {
+		return strconv.Itoa(int(value))
+	}
+	return sType
+}
+
+// GetEnumMap returns objectsearch type enum map to GUI
+func (value ObjectSearchType) GetEnumMap() *enummap.EnumMap {
+	return objectSearchTypeEnumMap
+}
+
+// Validate ObjectSearchSettings action (Implements ActionSettings interface)
+func (settings ObjectSearchSettings) Validate() ([]string, error) {
+	if settings.ID == "" {
+		return nil, errors.Errorf("%s no id defined", ActionObjectSearch)
+	}
+	switch settings.SearchSource {
+	case ObjectSearchSourceFromList:
+		if len(settings.SearchTerms) < 1 {
+			return nil, errors.Errorf("%s no search terms defined", ActionObjectSearch)
+		}
+	case ObjectSearchSourceFromFile:
+		if settings.Filename.IsEmpty() {
+			return nil, errors.Errorf("%s search source<%s> defined, but no searchtermsfile<%s> found or no filename set",
+				ActionObjectSearch, settings.SearchSource, settings.Filename)
+		}
+		if len(settings.Filename.Rows()) < 1 {
+			return nil, errors.Errorf("%s search source<%s> defined, but searchtermsfile<%s> contains no search terms",
+				ActionObjectSearch, settings.SearchSource, settings.Filename)
+		}
+	default:
+		return nil, errors.Errorf("%s source<%s> not supported", ActionObjectSearch, settings.SearchSource)
+	}
+	return nil, nil
+}
+
+// Execute ObjectSearchSettings action (Implements ActionSettings interface)
+func (settings ObjectSearchSettings) Execute(sessionState *session.State, actionState *action.State, connection *connection.ConnectionSettings, label string, reset func()) {
+	uplink := sessionState.Connection.Sense()
+
+	app, err := sessionState.CurrentSenseApp()
+	if err != nil {
+		actionState.AddErrors(err)
+		return
+	}
+
+	var genObj *enigma.GenericObject
+	var getDataSize func() (int, error)
+	switch settings.SearchType {
+	case ObjectSearchTypeField:
+		fieldList, err := app.GetFieldList(sessionState, actionState)
+		if err != nil {
+			actionState.AddErrors(err)
+			return
+		}
+		if fieldList.Layout() == nil || fieldList.Layout().FieldList == nil {
+			actionState.NewErrorf("no fieldlist layout")
+			return
+		}
+
+		listbox, err := fieldList.GetOrCreateSessionListboxSync(sessionState, actionState, uplink.CurrentApp.Doc, settings.ID)
+		if err != nil {
+			actionState.AddErrors(err)
+			return
+		}
+		genObj = listbox.EnigmaObject
+
+		getDataSize = func() (int, error) {
+			size := 0
+			err := sessionState.SendRequest(actionState, func(ctx context.Context) error {
+				listObject, err := listbox.ListObject(ctx)
+				if err != nil {
+					return errors.WithStack(err)
+				}
+
+				size = getListObjectDataSize(listObject)
+				return nil
+			})
+			return size, err
+		}
+	case ObjectSearchTypeDimension:
```

Comment:
yes, it also doesn't create up to 7 listboxes automatically as sdkexerciser does. Client would do that when actually clicking the "selection tool" , but more people used this to simulate doing an object search in the "current selections" and then this would be closer than the sdkexerciser was. I also couldn't get the master dimensions to show upp in QCS sense the way they where supposed to so things seems to have changed there. This felt like the "middle" way. But yes load will differ from sdkexerciser.